dashboard/lib/mega_section.rb

```diff
@@ -81,49 +90,75 @@ def self.create_teacher(email, password, name)
 
-    current_student = 0
-
-    # Create students in section
-    (1..options[:num_students]).each do
+    # Create students
+    students = []
+    (1..options[:num_students]).each do |i|
       # Choose random properties and create student
       age_min = options[:age_min]
       age_max_inclusive = options[:age_max_inclusive]
-      current_student += 1
 
-      name = format(SAMPLE_STUDENT_NAME_FORMAT, current_student)
-      age = rng.rand(age_min..age_max_inclusive)
-      student_user = create :student, name: name, age: age
+      # We give all sample students the same parent email to make it easy to look them
+      # up after we bulk-create them here.
+      students << build(:student,
+        name: format(SAMPLE_STUDENT_NAME_FORMAT, i),
+        age: rng.rand(age_min..age_max_inclusive),
+        email: "mega_student_#{i}@example.com",
+        parent_email: SAMPLE_STUDENT_PARENT_EMAIL
+      )
+    end
+    User.import! students
 
+    students = User.where(parent_email: SAMPLE_STUDENT_PARENT_EMAIL)
+
+    # Add students to section and simulate progress
+    followers = []
+    user_levels = []
+    teacher_feedbacks = []
+    students.each do |student_user|
       # Add student to section
-      create :follower, section: section, student_user: student_user
+      followers << build(:follower, section: section, student_user: student_user)
 
-      script.script_levels.each do |script_level|
-        # Create progress for this student on each level
-        user_level = create :user_level,
+      # Create progress for this student on each level
+      script_levels.each do |script_level|
+        user_levels << build(:user_level,
           user: student_user,
           script_id: script.id,
           level_id: script_level.levels.first.id,
           attempts: 1,
           best_result: ActivityConstants::BEST_PASS_RESULT
-        # Score each level for each student
-        create :teacher_score,
-          user_level_id: user_level.id,
-          teacher_id: section.teacher.id,
-          score: 100
+        )
+
         # Add teacher feedback for each level.
-        create :teacher_feedback,
+        teacher_feedbacks << build(:teacher_feedback,
           student_id: student_user.id,
           teacher_id: section.teacher.id,
           script_level_id: script_level.id,
           level_id: script_level.levels.first.id,
           comment: tiny_lipsum
+        )
       end
     end
+    Follower.import! followers
+    UserLevel.import! user_levels
+    TeacherFeedback.import! teacher_feedbacks
+
+    # Retrieve newly-created user levels so we can add teacher scores
+    teacher_scores = []
+    UserLevel.where(user: students).each do |user_level|
+      # Score each level for each student
+      teacher_scores << build(:teacher_score,
+        user_level_id: user_level.id,
+        teacher_id: section.teacher.id,
+        score: 100
+      )
+    end
+    TeacherScore.import! teacher_scores
```

Comment:
Same here: We're doing this outside of the students loop, and collecting up all TeacherScores for all the UserLevels for all the Students before we run `import!`